disease/etl/oncotree.py

```diff
@@ -0,0 +1,120 @@
+        """
+        self._extract_data()
+        self._load_meta()
+        self._transform_data()
+        self.database.flush_batch()
+        return []
+
+    def _download_data(self):
+        """Download Oncotree source data for loading into normalizer."""
+        logger.info('Downloading OncoTree...')
+        # get version for latest stable release
+        versions_url = f"{self._SRC_API_ROOT}versions"
+        versions = json.loads(requests.get(versions_url).text)
+        latest = [v['release_date'] for v in versions
+                  if v['api_identifier'] == 'oncotree_latest_stable'][0]
+        version = latest.replace('-', '_')
+
+        # download data
+        url = f'{self._SRC_API_ROOT}tumorTypes/tree?version=oncotree_{version}'
+        try:
+            response = requests.get(url, stream=True)
+        except requests.exceptions.RequestException as e:
+            logger.error(f'OncoTree download failed: {e}')
+            raise e
+        filename = self._data_path / f'oncotree_{version}.txt'
+        handle = open(filename, 'wb')
+        for chunk in response.iter_content(chunk_size=512):
+            if chunk:
+                handle.write(chunk)
+        self._version = version
+        logger.info('Finished downloading OncoTree')
+
+    def _load_meta(self):
+        """Load metadata"""
+        metadata = Meta(data_license="CC BY 4.0",
+                        data_license_url="https://creativecommons.org/licenses/by/4.0/legalcode",  # noqa F401
+                        version=self._version,
+                        data_url="http://oncotree.mskcc.org/#/home?tab=api",
+                        rdp_url=None,
+                        data_license_attributes={
+                            'non_commercial': False,
+                            'share_alike': False,
+                            'attribution': True
+                        })
+        params = dict(metadata)
+        params['src_name'] = SourceName.ONCOTREE.value
+        self.database.metadata.put_item(Item=params)
+
+    def _traverse_tree(self, disease_node):
+        """Traverse JSON tree and load diseases where possible.
+
+        :param Dict disease_node: node in tree containing info for individual
+            disease.
+        """
+        if disease_node.get('level', None) >= 2:
+            disease = {
+                "concept_id": f"{NamespacePrefix.ONCOTREE.value}:{disease_node['code']}",  # noqa: E501
+                "label": disease_node['name'],
+                "other_identifiers": [],
+            }
+            refs = disease_node.get('externalReferences', [])
+            for prefix, codes in refs.items():
+                if prefix == 'UMLS':
+                    normed_prefix = NamespacePrefix.UMLS.value
+                elif prefix == 'NCI':
+                    normed_prefix = NamespacePrefix.NCIT.value
+                else:
+                    raise Exception(f"Unrecognized prefix: {prefix}")
```

Comment:
Should we log this instead of raising an exception? I think in the past I've just logged, but want to make sure we're staying consistent in what we choose to do. If we do want to stick with raising exceptions, I'll just have to change in the other normalizers.